src/main/java/net/dv8tion/jda/internal/requests/restaction/pagination/ThreadChannelPaginationActionImpl.java

```diff
@@ -46,23 +49,27 @@ public EnumSet<PaginationOrder> getSupportedOrders()
         return EnumSet.of(PaginationOrder.BACKWARD);
     }
 
+    //Thread pagination supplies ISO8601 timestamps for some cases, see constructor
+    @Nonnull
     @Override
-    protected Route.CompiledRoute finalizeRoute()
+    protected String getPaginationLastEvaluatedKey(long lastId, ThreadChannel last)
     {
-        Route.CompiledRoute route = super.finalizeRoute();
-
-        final String limit = String.valueOf(this.limit.get());
-        final long last = this.lastKey;
+        if (useID)
+            return Long.toUnsignedString(lastId);
 
-        route = route.withQueryParams("limit", limit);
+        if (order == PaginationOrder.FORWARD && lastId == 0)
```

Comment:
This checks if the order is FORWARD, but the only supported order according to `getSupportedOrder` is `BACKWARDS` and infact if you manually change it to `FORWARD` the code doesn't work because it keeps fetching the same channels.